File: built-in plugins dsp/mb-eq-surgical.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbEqSurgicalPlugin: PluginDefinition = { id: 'mb-eq-surgical', slug: 'mb-eq-surgical', name: 'MB Surgical EQ', category: 'effect', type: 'eq', version: '1.0.0', description: 'Precision notch filtering', author: 'Max Booster', parameters: [{ id: 'freq', name: 'Frequency', type: 'float', defaultValue: 1000, minValue: 20, maxValue: 20000, automatable: true }, { id: 'gain', name: 'Gain', type: 'float', defaultValue: 0, minValue: -24, maxValue: 24, automatable: true }, { id: 'q', name: 'Q', type: 'float', defaultValue: 5, minValue: 0.5, maxValue: 20, automatable: true }], defaultPreset: { freq: 1000, gain: 0, q: 5 } };

export default MbEqSurgicalPlugin;

File: built-in plugins dsp/mb-eq-surgical.cpp
/**
 * MB Surgical EQ
 * Category : effect
 * Type     : eq
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Precision notch filtering
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_EQ_SURGICAL_H
#define MB_EQ_SURGICAL_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbEqSurgical : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-eq-surgical";
    static constexpr const char* PLUGIN_NAME    = "MB Surgical EQ";
    static constexpr const char* PLUGIN_TYPE    = "eq";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float freq = 1000f;  // range [20, 20000]
    float gain = 0f;  // range [-24, 24]
    float q = 5f;  // range [0.5, 20]
    };

    MbEqSurgical() = default;
    ~MbEqSurgical() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.freq = std::clamp(params.freq, 20f, 20000f);
        params.gain = std::clamp(params.gain, -24f, 24f);
        params.q = std::clamp(params.q, 0.5f, 20f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Surgical EQ
        return input;
    }
};

#endif // MB_EQ_SURGICAL_H
